contracts/contracts/eosio.system/src/voting.cpp

```diff
@@ -66,11 +66,14 @@ namespace eosiosystem {
             info.url             = url;
             info.location        = location;
             info.last_claim_time = ct;
+            info.last_expected_produced_blocks_update = ct;
          });
          _producers2.emplace( producer, [&]( producer_info2& info ){
             info.owner                     = producer;
             info.last_votepay_share_update = ct;
          });
+         const auto &voter = _voters.get(producer.value);
+         _gstate.total_producer_stake += voter.staked;
```

Comment:
I think we need to have logic to decrease total_producer_stake in unregprod() and add back if we reg again deactivated producer